api/src/fetch/loomio.js

```diff
@@ -0,0 +1,112 @@
+import cron from 'node-cron';
+import axios from 'axios';
+import { createInfo } from '../graphql/connectors';
+import driver from '../db/neo4jDriver';
+
+// TODO: Check encoding, make sure it's UTF-8?
+const loomioApi = axios.create({
+  baseURL: process.env.LOOMIO_API_BASE,
+  timeout: 10000,
+  headers: { 'User-Agent': 'Realities Agent', 'Accept-Encoding': 'gzip' },
+});
+
+// Set max results to ridiculously high number.
+const defaultParams = { per: 100000 };
+
+// Running without Loomio is okay, but suppress the errors.
+const checkEnvVars = () => {
+  const loomioVars = [
+    'LOOMIO_API_BASE',
+    'LOOMIO_SITE_BASE',
+    'LOOMIO_CRON_SCHEDULE',
+  ];
+  const errors = [];
+  loomioVars.forEach((s) => {
+    const v = process.env[s];
+    if (typeof v === 'undefined') {
+      errors.push(`WARNING: Missing variable "${s}" in .env file`);
+    } else if (!v) {
+      errors.push(`WARNING: Empty variable "${s}" in .env file`);
+    }
+  });
+  errors.forEach((e) => { console.error(e); });
+  return !errors.length;
+};
+
+// Loomio resourceName can be either 'discussions' or 'groups'.
+// The pathPrefix is 'd' for discussions and 'g' for groups.
+// Discussions have titles and groups have names.
+// The params are passed through to the axio GET request.
+const loomio = async (resourceName, pathPrefix, fieldName, params) => {
+  const url = `${resourceName}.json`;
+  const allParams = { params };
+
+  const result = [];
+  try {
+    const response = await loomioApi.get(url, allParams);
+    const objects = response.data[resourceName];
+
+    for (let i = 0; i < objects.length; i += 1) {
+      const { key } = objects[i];
+      const fieldValue = objects[i][fieldName];
+      const objUrl = `${process.env.LOOMIO_SITE_BASE}/${pathPrefix}/${key}`;
+      let info = '';
+      try {
+        /* eslint-disable no-await-in-loop */
+        info = await createInfo(driver, { title: fieldValue }, objUrl);
+      } catch (err) {
+        console.error(err);
```

Comment:
Errors here should be caught by the outer `catch`, so we probably don't need another try/catch here.